Misc/NEWS.d/next/Library/2023-12-02-12-55-17.gh-issue-112618.7_FT8-.rst

```diff
@@ -0,0 +1,2 @@
+Fix a caching bug relating to :data:`typing.Annotated`.
+``Annotated[str, True]`` is no longer equal to ``Annotated[str, 1]``.
```

Comment:
Oh, `Annotated[str, 1] == Annotated[str, True]` does still evaluate to `True`, even with this patch applied -- my mistake. But maybe that's correct, since `1 == True`, after all. What do you think?